### PR TITLE
An 2657/verify abis

### PR DESCRIPTION
**Pull request by austinFlipside**
- SPROC adds new ABIs to staging table in `bronze_public` schema
- creates a new model, `models/silver/user_submitted/silver__user_verified_abis.sql` to verify ABIs from community 
- verified ABIs are added to the `silver__abis` model
- `models/silver/streamline/decoder/streamline__backfill_decode_logs.sql` handles logs outside of last day of blocks, can be ran hourly
- `models/silver/streamline/decoder/streamline__decode_logs_realtime.sql` handles logs within last day of blocks, will run with core job
- most file changes are history files being moved into their own folder

### CODE REVIEW

File: models/silver/streamline/decoder/streamline__backfill_decode_logs.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        WHERE
+            block_number < (
+                SELECT
+                    block_number
+                FROM
+                    look_back
+            )
+    )
+LIMIT
+    15000000
```
Comment (juls858) [inline models/silver/streamline/decoder/streamline__backfill_decode_logs.sql:54]: why is the limit needed?

File: models/silver/user_submitted/silver__user_verified_abis.sql [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        recent_logs
+),
+successful_abis AS (
+    SELECT
+        abi_address,
+        SUM(
+            CASE
+                WHEN decoded_event_name IS NOT NULL THEN 1
+            END
+        ) AS successful_rows,
```
Comment (juls858) [inline models/silver/user_submitted/silver__user_verified_abis.sql:167]: coalesce would be better here

File: models/silver/silver__abis.sql [hunk 3]
```diff
@@ -31,26 +31,80 @@ WHERE
...
+        user_abis
+)
 SELECT
-    contract_address,
-    block_number,
-    DATA,
-    _INSERTED_TIMESTAMP,
-    metadata,
-    VALUE
+    *
```
Comment (juls858) [inline models/silver/silver__abis.sql:106]: should be expanded

File: models/silver/user_submitted/silver__user_verified_abis.sql [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        COUNT(*) AS total_rows,
+        successful_rows / total_rows AS success_rate
+    FROM
+        decoded_logs
+    GROUP BY
+        1
+)
+SELECT
+    contract_address,
+    abi,
```
Comment (juls858) [inline models/silver/user_submitted/silver__user_verified_abis.sql:177]: can we add a hash of the abi here? how are we verifying duplicate abis?

File: models/silver/user_submitted/silver__user_verified_abis.sql [hunk 5]
```diff
@@ -0,0 +1,190 @@
+{{ config (
+    materialized = "incremental",
+    unique_key = "contract_address",
```
Comment (juls858) [inline models/silver/user_submitted/silver__user_verified_abis.sql:3]: is this correct?

**Review by juls858 — CHANGES_REQUESTED** [review]
requested some changes, see the comments

**Review by juls858 — CHANGES_REQUESTED** [review]
requested some changes, see the comments

File: models/silver/streamline/decoder/streamline__backfill_decode_logs.sql (hunk 1 above)
Comment (austinFlipside) [reply models/silver/streamline/decoder/streamline__backfill_decode_logs.sql:54]: guarantees under 15 mins for select, this around 10 mins

File: models/silver/user_submitted/silver__user_verified_abis.sql (hunk 5 above)
Comment (austinFlipside) [reply models/silver/user_submitted/silver__user_verified_abis.sql]: updated

File: models/silver/user_submitted/silver__user_verified_abis.sql (hunk 4 above)
Comment (austinFlipside) [reply models/silver/user_submitted/silver__user_verified_abis.sql:178]: added

**Review by juls858 — APPROVED** [review]

**Review by drethereum — APPROVED** [review]

**Pull request merged**